aexpect/client.py

```diff
@@ -753,21 +762,34 @@ def read_nonblocking(self, internal_timeout=None, timeout=None):
         poller = select.poll()
         poller.register(expect_pipe, select.POLLIN)
         data = ""
+        read = 0
         while True:
             try:
                 poll_status = poller.poll(internal_timeout)
             except select.error:
-                return data
+                return read, data
             if poll_status:
-                new_data = os.read(expect_pipe, 1024).decode(self.encoding,
-                                                             "ignore")
-                if not new_data:
-                    return data
-                data += new_data
+                raw_data = os.read(expect_pipe, 1024)
+                if not raw_data:
+                    return read, data
+                read += len(raw_data)
```

Comment:
it sound like read  need be calculated based on  return value of  len(raw_data.decode(self.encoding, "ignore")), otherwise read is always consistent with raw_data.